### DISCUSSION TITLE
Decorator for custom events

### DISCUSSION THREAD

**Discussion by hakimio** [dxfrontier/cds-ts-dispatcher#15] [Q&A] [answered]
Is there any decorator which can be used for handling custom events like the following:
```ts
await this.srv.emit('OrderedBook', { book, quantity, buyer: req.user.id });
```

**Comment by hakimio** [comment]
Is it just `@UnboundActions()`?

**Reply by dragolea** [reply]
Hi,
@UnboundActions can be used to have Unbound actions ( actions / functions that are not bound to an entity )

**Comment by hakimio** [comment]
@dragolea how do you handle custom events?

**Reply by dragolea** [reply]
Hi,
You can use the UnboundActions decorator to handle custom actions for a service, those actions which are not bound to an entity but to the service itself.

Hope I understood correctly your question, if not plese just reply again.

Sorry for delay,
Christmas vacation.

**Comment by dragolea** [comment, marked as answer]
Regarding this question with the emit

await this.srv.emit('OrderedBook', { book, quantity, buyer: req.user.id });

You can use @Inject(SRV) private readonly srv : Service to access the CDS application service which can be used yo have the emit method

But currently we do not have a decorator for emit, we plan to have this feature in the near future.

**Reply by hakimio** [reply]
So, it's not possible to listen to events emitted this way with `@UnboundActions()` controller?
How do you handle this type of events in your own application? Do you just manually subscribe with `this.srv.on()`?

**Reply by dragolea** [reply]
Hi,

That decorator can be used only for unbound actions and functions which are not bound to an actual entity

Maybe you can provide me a code snippet and an actual requirement of what you want to achieve, so we can have in our coming release that feature.

P.S. please open an issue for a new Requirement, we will be more than. happy to help you.
